src/Stateless/Graph/UmlDotGraphStyle.cs

```diff
@@ -0,0 +1,179 @@
+            es.AddRange(state.ExitActions.Select(act => "exit / " + act));
+
+            f += String.Join("\\n", es);
+
+            f += "\"];\n";
+
+            return f;
+        }
+
+        /// <summary>
+        /// Format all transitions for a graph
+        /// </summary>
+        /// <param name="transitions"></param>
+        /// <returns></returns>
+        override internal List<string> FormatAllTransitions(List<Stateless.Graph.Transition> transitions)
+        {
+            List<string> lines = new List<string>();
+
+            foreach (var transit in transitions)
+            {
+                string line = null;
+                if (transit is StayTransition stay)
+                {
+                    if (!stay.ExecuteEntryExitActions)
+                    {
+                        line = FormatOneTransition(stay.SourceState.NodeName, stay.Trigger.UnderlyingTrigger.ToString(),
+                            null, stay.SourceState.NodeName, stay.Guards.Select(x => x.Description));
+                    }
+                    else if (stay.SourceState.EntryActions.Count() == 0)
+                    {
+                        line = FormatOneTransition(stay.SourceState.NodeName, stay.Trigger.UnderlyingTrigger.ToString(),
+                            null, stay.SourceState.NodeName, stay.Guards.Select(x => x.Description));
+                    }
+                    else
+                    {
+                        // There are entry functions into the state, so call out that this transition
+                        // does invoke them (since normally a transition back into the same state doesn't)
+                        line = FormatOneTransition(stay.SourceState.NodeName, stay.Trigger.UnderlyingTrigger.ToString(),
+                            stay.SourceState.EntryActions, stay.SourceState.NodeName, stay.Guards.Select(x => x.Description));
+                    }
+                }
+                else
+                {
+                    if (transit is FixedTransition fix)
+                    {
+                        line = FormatOneTransition(fix.SourceState.NodeName, fix.Trigger.UnderlyingTrigger.ToString(),
+                            fix.DestinationEntryActions.Select(x => x.Method.Description),
+                            fix.DestinationState.NodeName, fix.Guards.Select(x => x.Description));
+                    }
+                    else
+                    {
+                        if (transit is DynamicTransition dyn)
+                        {
+                            line = FormatOneTransition(dyn.SourceState.NodeName, dyn.Trigger.UnderlyingTrigger.ToString(),
+                                dyn.DestinationEntryActions.Select(x => x.Method.Description),
+                                dyn.DestinationState.NodeName, new List<string> { dyn.Criterion });
+                        }
+                        else
+                            throw new System.Exception("Unexpected transition type");
```

Comment:
Could we use a more specific exception type? If you throw such a general exception type in a library or framework, it forces consumers to catch all exceptions, including unknown exceptions that they do not know how to handle.